### DISCUSSION TITLE
Consider adding the repost record URI to app.bsky.feed.defs#reasonRepost

### DISCUSSION THREAD

**Discussion by edavis** [bluesky-social/atproto#2702] [Bluesky Lexicons]
I was recently playing around with taking my `getTimeline` results and creating a custom feed out of it. It worked great except for one thing:

For posts included because one of my follows reposted it, the `reason` field only tells me _who_ reposted it and when it was indexed. If there was a `reason.uri` field with the URI of the repost record itself, I could attach that when generating my skeleton feedgen response. And with that in place, it would be hydrated to include the little "Reposted by XXX" indicator above the post.

I imagine the intention here was "you're making a feedgen, track and include those repost record URIs yourself" which I can't really argue with. My thinking, though, is using `getTimeline` gets me 99% toward my destination. Including the repost record URI would put me at 100%. So I wanted to see if there was any interest in this approach before deciding what to do next.

Thanks!

**Comment by snarfed** [comment]
Yes! This is a long-standing known issue, I'd dearly like it too. We're already tracking it (hopefully) in https://github.com/bluesky-social/atproto/issues/1811
